tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -55,18 +55,40 @@ struct dt;
 /// \endcond
 
 namespace GeneralizedHarmonic {
+namespace gauges {
 namespace DampedHarmonicGauge_detail {
 // The `detail` functions below are forward-declared to enable their independent
 // testing
 template <size_t SpatialDim, typename Frame, typename DataType>
-Scalar<DataType> weight_function(
+void spatial_weight_function(gsl::not_null<Scalar<DataType>*> weight,
+                             const tnsr::I<DataType, SpatialDim, Frame>& coords,
+                             double sigma_r) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spacetime_deriv_of_spatial_weight_function(
+    gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> d4_weight,
+    const tnsr::I<DataType, SpatialDim, Frame>& coords, double sigma_r,
+    const Scalar<DataType>& weight_function) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> spatial_weight_function(
     const tnsr::I<DataType, SpatialDim, Frame>& coords,
-    double sigma_r) noexcept;
+    const double sigma_r) noexcept {
+  Scalar<DataType> weight{};
+  spatial_weight_function(make_not_null(&weight), coords, sigma_r);
+  return weight;
+}
```

Comment:
why is return by value only in the test?

Reply:
Because the return-by-value is easier to use, but should only be used in the test since in the production code we want to reduce allocations. By not having it in `src` it's rather difficult to use in the production code :)

Reply:
Perhaps there should be a comment here saying that we put return-by-value only in the test on purpose?